### DISCUSSION TITLE
Welcome to libretranslate-java Discussions!

### DISCUSSION THREAD

**Discussion by dynomake** [dynomake/libretranslate-java#1] [Announcements]
👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Plyha** [comment]
Hello!
Thank you very much for your work!

Will be auto-detection of the text?

**Comment by dynomake** [comment]
yes, it will be soon

**Comment by dynomake** [comment]
Done! If you don't know which language to translate from, use:
```java
Translator.translate("to", "text");
```
Usage example:
```java
System.out.println(Translator.translate("en", "Для примера, переведем текст с русского на английский и выведем в консоль:"));
```

**Comment by Plyha** [comment]
Excellent!

How can I get a list of all supported languages?

**Comment by dynomake** [comment]
Done! View all available languages: https://github.com/suuft/libretranslate-java/blob/master/src/main/java/net/suuft/libretranslate/Language.java

Now you can specify Enum, example:
```java
Translator.translate(Language.RUSSIAN, Language.ENGLISH, "Для примера, переведем текст с русского на английский и выведем в консоль:");
```
Also, if you do not know the language from you need to translate:
```java
Translator.translate(Language.ENGLISH, "Для примера, переведем текст с русского на английский и выведем в консоль:");
```

**Comment by Plyha** [comment]
Cool!

**Comment by Plyha** [comment]
Can you also add a translate method with a String parameter?
In my case, this would be very useful.

**Reply by dynomake** [reply]
The translator does not provide for this. I will do it tomorrow (if you dont mind)

**Reply by Plyha** [reply]
yes, sure.  Thanks

**Comment by Plyha** [comment]
I would like to parse ENUM + add my values ​

**Reply by dynomake** [reply]
Do you want to add your values to languages ENUM?

**Reply by Plyha** [reply]
Yes, in my application there should be a list of languages ​​+ NONE value.  This is necessary so as not to translate the text.

**Reply by dynomake** [reply]
okay, i added the NONE value to languages list

**Comment by Plyha** [comment]
return "Falled translate!";

I think it should be "Fa**i**led translate!"

**Reply by dynomake** [reply]
If language is NONE?

**Reply by Plyha** [reply]
I mean there is a typo

**Reply by dynomake** [reply]
upss, i understand
